### PR TITLE
WAITP-1364 Add map_overlay.config to types package

### PR DESCRIPTION
**Pull request by EMcQ-BES**
### Changes:

- Add `map_overlay.config` definition to the types package
- Move existing enumerated string definitions into types packages
- Update `tupaia-web` to use the new type definitions

### CODE REVIEW

File: packages/tupaia-web/src/types/mapOverlay.d.ts [hunk 1]
```diff
@@ -1,33 +0,0 @@
-/*
```
Comment (tcaiger) [inline packages/tupaia-web/src/types/mapOverlay.d.ts:1]: Great to be able to delete this

Comment (tcaiger) [reply packages/tupaia-web/src/types/mapOverlay.d.ts:1]: Do we need to delete the line from types/index.d.ts as well?

File: packages/ui-map-components/src/components/Markers/markerIcons.tsx [hunk 2]
```diff
@@ -250,71 +250,71 @@ const FadedCircle = ({ color, scale = 1 }: IconProps) => {
 };
 
 const icons = {
-  pin: {
+  [IconKey.PIN]: {
```
Comment (tcaiger) [inline packages/ui-map-components/src/components/Markers/markerIcons.tsx:253]: Nice - this is a great improvement

**Review by tcaiger — APPROVED** [review]
Looks good to me. It's a huge improvement to be replacing all these random const with types from the types package. Just one question about the index file in tupaia-web/src/types

File: packages/tupaia-web/src/types/mapOverlay.d.ts (hunk 1 above)
Comment (EMcQ-BES) [reply packages/tupaia-web/src/types/mapOverlay.d.ts:1]: Good catch, removed

**Pull request merged**